spirekey/tests/utils/merkle.test.ts

```diff
@@ -0,0 +1,80 @@
+import { getMerkleProofWith, getRootHashWith } from '@/utils/merkle';
+import { describe, expect, it } from 'vitest';
+
+describe('Merkle', () => {
+  const hashMock = (value: string) => `H(${value})`;
+  const getRootHash = getRootHashWith(hashMock);
+  const getMerkleProof = getMerkleProofWith(hashMock);
+  describe('When getting a merkle root hash', () => {
+    describe('And we have 4 entries', () => {
+      it('should get the hash based on `H(H(A,B),H(C,D))`', () => {
+        const hashes = ['A', 'B', 'C', 'D'];
+        expect(getRootHash(hashes)).toEqual('H(H(A,B),H(C,D))');
+      });
+    });
+    describe('And we have 3 entries', () => {
+      it('should get the hash based on `H(H(A,B),H(C,0))`', () => {
+        const hashes = ['A', 'B', 'C'];
+        expect(getRootHash(hashes)).toEqual('H(H(A,B),H(C,0))');
+      });
+    });
+    describe('And we have 5 entries', () => {
+      it('should get the hash based on `H(H(H(A,B),H(C,D)),H(H(E,0),H(0,0)))`', () => {
+        const hashes = ['A', 'B', 'C', 'D', 'E'];
+        expect(getRootHash(hashes)).toEqual(
+          'H(H(H(A,B),H(C,D)),H(H(E,0),H(0,0)))',
+        );
+      });
+    });
+  });
+  describe('When getting the merkle proof', () => {
+    describe('And we have 4 entries', () => {
+      describe('And we want to have the proof for A', () => {
+        it('should get [B, H(C,D)]', () => {
+          const hashes = ['A', 'B', 'C', 'D'];
+          expect(getMerkleProof('A', hashes)).toEqual(['B', 'H(C,D)']);
```

Comment:
For the elements of the proof, in addition to the hash of each step, don't we also need to indicate whether the hash is applied to the running result from the left/right?

e.g.

```
expect(getMerkleProof('A', ['A', 'B', 'C').toEqual(
  [ { side: "Right", hash: "B" },
    { side: "Right", hash: H('C', 'D') }
  ])

expect(getMerkleProof('C', ['A', 'B', 'C']).toEqual(
  [ { side: "Left", hash: H('A', 'B') } ])
```

Reply:
As we discussed in slack, we are moving to use the more common scheme described here: https://datatracker.ietf.org/doc/html/rfc9162#name-merkle-inclusion-proofs-2 with the exclusion of `log_id` represented as:

```json
{
  "tree_size": number,
  "leaf_index": number,
  "inclusion_path": string[]
}
```

Reply:
Slack update: we're back to adding `side` info to the proof steps :D